src/server/pfs/server/api_server.go

```diff
@@ -704,9 +717,27 @@ func (a *apiServer) DeleteFile(ctx context.Context, request *pfs.DeleteFileReque
 }
 
 func (a *apiServer) Version(version int64) error {
-	a.versionLock.Lock()
-	defer a.versionLock.Unlock()
-	a.version = version
+	func() {
+		a.versionLock.RLock()
+		defer a.versionLock.RUnlock()
+
+		a.versionChanLock.Lock()
+		defer a.versionChanLock.Unlock()
+
+		if !a.versionChanClosed[a.version] {
+			close(a.versionChans[a.version])
+			a.versionChanClosed[a.version] = true
+			a.versionChans[version] = make(chan struct{})
+		}
```

Comment:
I don't think you need the `versionChanClosed` map because I think `Version` should only get called once with each value.